datalad/distribution/publish.py

```diff
@@ -15,108 +15,94 @@
+from .dataset import datasetmethod
+from .install import get_containing_subdataset
+
 
 lgr = logging.getLogger('datalad.distribution.publish')
 
 
 class Publish(Interface):
-    """Publish (e.g. to a web server) a dataset(s)
+    """Publish a dataset (e.g. to a web server)
+
+    This makes the current state of a dataset available to a sibling of the
+    dataset. That sibling needs to be known to the dataset before.
+    `create_publication_target` commands are meant to set up such a sibling, but
+    generally you can publish to any sibling added to a dataset.
+    Publishing may or may not include subdatasets.
+    By default file handles are published without their actual content,
+    but they can be published including the content, of course.
 
-    This is basic implementation for testing purposes
     """
+    # TODO: Figure out, how to tell about tracking branch/upstream
+    #      (and the respective remote)
+    #      - it is used, when no destination is given
+    #      - it is configured to be the given destination, if there was no
+    #        upstream set up before, so you can use just "datalad publish" next
+    #        time.
 
     _params_ = dict(
         dataset=Parameter(
             args=("-d", "--dataset"),
-            doc="""specify the dataset to perform the publish operation on.  If
-            no dataset is given, an attempt is made to identify the dataset
-            based on the current working directory and/or the `path` given""",
+            metavar='DATASET',
+            doc="""specify the dataset to publish. If no dataset is given, an
+            attempt is made to identify the dataset based on the current
+            working directory""",
             constraints=EnsureDataset() | EnsureNone()),
         dest=Parameter(
             args=("dest",),
-            doc="""url, local path, or sibling name identifying the publication
-            target""",
+            metavar='DEST',
+            doc="""sibling name identifying the publication target. If no
+            destination is given an attempt is made to identify the target based
+            on the dataset's configuration.""",
+            # TODO: See TODO at top of class!
             nargs="?",
             constraints=EnsureStr() | EnsureNone()),
-        path=Parameter(
-            args=("path",),
-            doc="path/name of the dataset component to publish",
-            nargs="*",
-            constraints=EnsureStr() | EnsureNone()),
         recursive=Parameter(
             args=("-r", "--recursive"),
             action="store_true",
-            doc="recursively publish all components of the dataset"),
+            doc="recursively publish all subdatasets of the dataset"),
         with_data=Parameter(
             args=("--with-data",),
-            doc="shell pattern",
+            metavar='PATH',
+            doc="path(s) to file handle(s) to publish including their actual "
+                "content",
```

Comment:
is that full names? or globs or regexps?
I wonder if it wouldn't be better to stay inline with 'git commit', 'git annex copy' and having `publish [DEST [PATHs]]` instead of `publish --with-data "crumble multiple entries into a single arg so globs wouldn't do it at shell level"`.  That would require of cause having only a single `DEST`, now it is specified for multiple but which one would be more common use case?